interfaces/smtp/v0/schema.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2023 Canonical
+# See LICENSE file for licensing details.
+"""This file defines the schema for the provider side of the smtp interface.
+
+It exposes one interfaces.schema_base.DataBagSchema subclass called:
+- ProviderSchema
+
+Examples:
+    ProviderSchema:
+        unit: <empty>
+        app: {"smtp":
+                 {
+                    "host": "example.smtp",
+                    "port": "25",
+                    "user": "example_user",
+                    "password_id": "01548499c9233d4612352c989162d940f6a9e6f6d5cc058dfcf66f51575e09c2",
+                    "auth_type": "plain",
+                    "transport_security": "tls",
+                    "domain": "domain",
+                }
+             }
+"""
+from enum import Enum
+from interface_tester.schema_base import DataBagSchema
+from pydantic import BaseModel, Field
+from typing import Optional
+
+
+class TransportSecurity(str, Enum):
+    """Represent the transport security values.
+
+    Attributes:
+        NONE: none
+        STARTTLS: starttls
+        TLS: tls
+    """
+
+    NONE = "none"
+    STARTTLS = "starttls"
+    TLS = "tls"
+
+
+class AuthType(str, Enum):
+    """Represent the auth type values.
+
+    Attributes:
+        NONE: none
+        NOT_PROVIDED: not_provided
+        PLAIN: plain
+    """
+
+    NONE = "none"
+    NOT_PROVIDED = "not_provided"
+    PLAIN = "plain"
+
+class SmtpProviderData(BaseModel):
+    host: str = Field(
+        ...,
+        min_length=1,
+        description="SMTP host.",
+        title="Host",
+        examples=["example.smtp"],
+    )
+    port: int = Field(None,
+        ge=1,
+        le=65536,
+        description="SMTP port.",
+        title="Port",
+        examples=[25],
+    )
+    user: Optional[str] = Field(
+        description="SMTP user.",
+        title="User",
+        examples=["some_user"],
+    )
+    password_id: Optional[str] = Field(
+        description="SMTP password.",
```

Comment:
```suggestion
        description="Juju Secret ID that resolves to a SMTP password.",
        ...
        examples=["secret:123213123123123123123"]
```

Reply:
Fixed, thx

Reply:
@arturo-seijas , please resolve the conversation once the fix is actually made.  This applies at multiple places here.

Reply:
The fit has been pushed. That's why I resolved. Do you see anything weird?